anomalib/data/folder.py

```diff
@@ -141,7 +142,7 @@ class FolderDataset(AnomalibDataset):
     """Folder dataset.
 
     Args:
-        task (str): Task type. (classification or segmentation).
+        task (TaskType): Task type. (classification, detection or segmentation).
```

Comment:
For consistency, following your changes above.
```suggestion
        task (TaskType): Task type. (``classification``, ``detection`` or ``segmentation``).
```